packages/web3-eth-accounts/src/account.ts

```diff
@@ -55,17 +80,146 @@ export const privateKeyToAccount = (
+    privateKey: string;
+    signTransaction: () => boolean; // From 1.x
+    sign: () => boolean;
+    encrypt: (a: string,b: string) => Promise<V3Keystore>;
+}>=> {
+    if(!(typeof password === 'string')) throw new Error('');
+    
+    const json = (typeof v3Keystore === 'object') ? v3Keystore : JSON.parse(nonStrict ? v3Keystore.toLowerCase() : v3Keystore) as V3Keystore;
+
+    validateKeyStore(json)
+
+    if (json.version !== 3) throw new Error('Not a valid V3 wallet');
+
+    let derivedKey;
+    if (json.crypto.kdf === 'scrypt') {
+        const kdfparams = json.crypto.kdfparams as ScryptParams;
+
+        // TODO support progress reporting callback
+        kdfparams.salt = typeof kdfparams.salt === 'string' ? Buffer.from(kdfparams.salt, 'hex') : kdfparams.salt
+
+        derivedKey = scryptSync(Buffer.from(password), Buffer.from(kdfparams.salt), kdfparams.n, kdfparams.p, kdfparams.r, kdfparams.dklen);
+    } else if (json.crypto.kdf === 'pbkdf2') {
+        const kdfparams: PBKDF2SHA256Params = json.crypto.kdfparams as PBKDF2SHA256Params;
+
+        if (kdfparams.prf !== 'hmac-sha256') {
+            throw new Error('Unsupported parameters to PBKDF2');
+        }
+
+        derivedKey = pbkdf2Sync(Buffer.from(password), Buffer.from(kdfparams.salt), kdfparams.c, kdfparams.dklen, 'sha256');
+    } else {
+        throw new Error('Unsupported key derivation scheme');
+    }
+
+    const ciphertext = hexToBytes(`0X${json.crypto.ciphertext}`);
+    const mac = sha3Raw(Buffer.from([...derivedKey.slice(16, 32), ...ciphertext])).replace('0x', '');
+    
+    if (mac !== json.crypto.mac) {
+        throw new Error('Key derivation failed - possibly wrong password');
+    }
+
+    const seed = await createDecipheriv(Buffer.from(json.crypto.ciphertext, 'hex'), derivedKey.slice(0, 16), Buffer.from(json.crypto.cipherparams.iv, 'hex'));
+  
+    return privateKeyToAccount(Buffer.from(seed));
+};
+
+/**
+ * encrypt a private key given a password, returns a V3 JSON Keystore
+ * https://github.com/ethereum/wiki/wiki/Web3-Secret-Storage-Definition
+ */
+export const encrypt = async (privateKey: string, password: string | Buffer, options?: CipherOptions): Promise<V3Keystore> => {
+    const account = privateKeyToAccount(privateKey);
+
+    // if given salt or iv is a string, convert it to a Uint8Array
+    let salt;
+    if (options?.salt){
+        salt = typeof options.salt === 'string' ? Buffer.from(options.salt, 'hex'): options.salt
+    } else {
+        salt = randomBytes(32);
+    }
+    validateBytesInput(salt);
+
+    let iv;
+    if (options?.iv){
+        iv = typeof options.iv === 'string' ? Buffer.from(options.iv, 'hex'): options.iv
+    } else {
+        iv = randomBytes(16)
+    }
+    validateBytesInput(iv);
+
+    const kdf = options?.kdf ?? 'scrypt';
+
+    let derivedKey;
+    let kdfparams: ScryptParams | PBKDF2SHA256Params;
+
+    // derive key from key derivation function
+	if (kdf === 'pbkdf2') {
+        kdfparams = {
```

Comment:
```suggestion
        kdfParams = {
```

Reply:
i stuck with all lowercase because `kdfparams` is how it is in 1.x, otherwise it would be considered a breaking change